### ISSUE TITLE
dot-notation: enabling the rule without type-aware mode is accepted but silently runs nothing — ESLint core flags `obj["A"]` syntactically

### ISSUE THREAD

**Issue by ahmedelgabri** [oxc-project/oxc#25614]
## Summary

oxlint's only implementation of `dot-notation` is the type-aware `typescript/dot-notation` (💭, requires `oxlint-tsgolint`). Enabling `"dot-notation": "error"` in `.oxlintrc.json` without type-aware mode passes config validation but silently activates zero rules — computed member accesses with plain string-literal properties (e.g. `obj["A"]`), the most basic case the rule exists to catch, are never reported. ESLint's core `dot-notation` is purely syntactic and flags them with no type information, so a migrated config silently loses all of these diagnostics.

## oxlint version

1.78.0; originally observed on 1.69.0, re-verified on 1.72.0 and 1.78.0.

## Reproduction

`repro.ts`:

```ts
const layout = {nodes: {} as Record<string, {w: number}>}
layout.nodes['A']
layout.nodes['sub1']
```

`.oxlintrc.json`:

```json
{
	"plugins": [],
	"categories": {"correctness": "off"},
	"rules": {"dot-notation": "error"}
}
```

Run:

```
oxlint -c .oxlintrc.json --disable-nested-config repro.ts
```

## Expected

Two diagnostics, matching ESLint core's `dot-notation`:

```
repro.ts:2:14  error  ["A"] is better written in dot notation.    dot-notation
repro.ts:3:14  error  ["sub1"] is better written in dot notation. dot-notation
```

String-literal property names in computed member access (`MemberExpression` with `computed: true` and a string `Literal` property that is a valid identifier) are exactly what `dot-notation` is meant to flag, and the core ESLint rule needs no type information for this. If oxlint intends this rule to be type-aware-only, enabling it without type-aware mode should at least warn instead of silently no-oping.

## Actual

oxlint 1.78.0 reports nothing (exit 0). `--format=json` shows the rule never ran:

```
{ "diagnostics": [], "number_of_files": 1, "number_of_rules": 0, ... }
```

The rule name is recognized — a bogus name fails config parsing with `Rule 'totally-bogus-rule-xyz' not found in plugin 'eslint'`, and `{"rules": {"no-eval": "error"}}` in the identical config shape runs and reports — but `dot-notation` activates zero rules. An `/* eslint-disable dot-notation */` on the same file is even flagged as `Unused eslint-disable directive (no problems were reported)`.

## Cross-checked

| Tool | Result |
| --- | --- |
| ESLint 7.23.0, core `dot-notation` (`@typescript-eslint/parser` 6.21) | 2 errors (2:14, 3:14) |
| ESLint 9.39.4 flat config, core `dot-notation` (`typescript-eslint` 8.62.1 parser, no `parserOptions.project`) | 2 errors (2:14, 3:14) |
| oxlint 1.72.0 | 0 diagnostics, `number_of_rules: 0` |
| oxlint 1.78.0 | 0 diagnostics, `number_of_rules: 0` |
